### ISSUE TITLE
Uni-HamGNN: clarifications on training-set composition and the OpenMX label protocol (for accurate description and consistent reference calculations)

### ISSUE THREAD

**Issue by hjunxi22** [QuantumLab-ZY/HamGNN#114]
Hi, and thank you for releasing Uni-HamGNN. We are using the released `predictor.pkl` (Zenodo 15568557) with OpenMX-generated `graph_data.npz` inputs for inorganic halide perovskites. We are preparing a methods section that describes the pretrained model, and we also run our own OpenMX reference calculations for comparison, which we would like to match the training protocol as closely as possible. A few details are not stated in the paper/SI (arXiv:2504.19586 / Nat. Mach. Intell. 2026) and we could not find them in this repository, so we would be grateful for clarification.

**Training set**

1. Structure selection from the Materials Project: were any filters applied (e.g., maximum atoms per cell, energy above hull, exclusion of magnetic or f-element systems), and which MP release was used?
2. Sec. 2.2 mentions 44,000 non-SOC structures for H0, while the Introduction mentions 40,000. Which is the exact number? Is the ~10,000-structure SOC set a subset of the non-SOC set, and was it drawn randomly or weighted toward heavy elements?
3. Does each structure contribute a single relaxed MP geometry (i.e., no perturbed or MD configurations)?
4. Roughly, what is the distribution of atoms per cell in the training set (median / maximum)? If a list of MP IDs for the training and test splits could be shared, that would let us check overlap with our own test systems.
5. In Fig. 3b ("Element count in the whole training dataset"), are the numbers atom counts or structure counts?

**OpenMX label protocol**

6. Exchange-correlation functional (we assume GGA-PBE, as for the CPL 2024 non-SOC model), OpenMX version, and the PAO/VPS database (DFT_DATA19?).
7. Was the 6×6×6 Monkhorst-Pack grid used for all cells regardless of cell size, or scaled with the cell dimensions?
8. `scf.criterion`: the paper states 1.0e-8 Hartree, while the readme template uses 1.0e-7. Which was used for the training labels? Likewise, were `scf.ElectronicTemperature 100` and `rmm-diis` mixing from the readme SOC template the training settings?
9. Can we confirm that the training labels used exactly the PAO/VPS choices in `DFT_interfaces/openmx/utils.py` (`PAO_dict` / `PBE_dict`, i.e., basis_def_26 including the "quick" bases for P/S/Cl/Ar discussed in #63), with `scf.SpinPolarization nc` and `scf.SpinOrbit.Coupling on` for the SOC labels?

Thanks again for your time.

**Comment by QuantumLab-ZY** [comment]
# Response to Questions

Thanks for the careful reading—answers are provided below, following your numbering.

## Training Set

1. The candidate pool was the Materials Project structure set released with the M3GNet universal potential (Chen & Ong, *Nature Computational Science*, 2022; MP data, 2021 release).

   We applied no filters of our own—no hull-energy, atom-count, magnetic, or f-element exclusions. The dataset was effectively reduced by the following practical constraints:

   1. The OpenMX PAO/VPS database does not cover all elements, so structures containing unsupported elements could not be computed.
   2. A fraction of the DFT runs did not reach SCF convergence.
   3. Computing all of MP would have been too costly.

   In total, more than approximately 50,000 structures were computed: 44,000 for training and approximately 5,500 for testing.

2. **44,000** (Section 2.2) is the exact number of non-SOC training structures; the **40,000** in the Introduction is a rounded figure. The approximately 10,000 SOC structures are a subset of the non-SOC set.

   All splits—including the train/test split and the SOC subset—were drawn uniformly at random, with no weighting toward heavy elements.

3. Yes. Each structure contributes a single relaxed MP geometry; no perturbed or molecular dynamics (MD) configurations were included.

4. We do not have the MP-ID lists in a shareable form at the moment and cannot curate them right now—sorry.

   Since all splits were drawn uniformly at random from the same pool, we do not expect systematic overlap with your test systems. If specific structures matter for your comparison, feel free to point them out, and we can check them individually.

5. The counts in **Figure 3b** are atom counts—that is, the total number of atoms of each element summed over the dataset—not structure counts.

## OpenMX Label Protocol

6. The calculations used:

   - **Exchange-correlation functional:** GGA-PBE
   - **Software:** OpenMX 3.9
   - **PAO/VPS database:** DFT_DATA19

7. A fixed **6 × 6 × 6 Monkhorst–Pack grid** was used for all cells, regardless of cell size, as stated in the paper.

8. The training calculations used:

   ```text
   scf.ElectronicTemperature 100
   ```

   Both `1.0e-7` and `1.0e-8` Hartree appear as `scf.criterion` values across the training runs. Once SCF convergence is reached, this difference has a negligible effect on the resulting Hamiltonians, so either value is consistent with the training labels.

   The calculations used the `rmm-diisk` mixing method, which likewise does not affect the converged Hamiltonian values.

9. Confirmed: the labels used exactly the PAO/VPS choices specified in:

   ```text
   DFT_interfaces/openmx/utils.py
   ```

   Specifically, they used `PAO_dict` and `PBE_dict`—that is, `basis_def_26`, including the “quick” basis sets for P, S, Cl, and Ar discussed in issue `#63`.

   The SOC labels used the following settings:

   ```text
   scf.SpinPolarization NC
   scf.SpinOrbit.Coupling on
   ```
